src/lamp_py/performance_manager/flat_file.py

```diff
@@ -59,49 +70,65 @@ def db_service_dates_to_datetimes(df: pandas.DataFrame) -> Set[datetime]:
             for service_date in df["service_date"]
         )
 
-    def filepaths_to_datetimes(filepaths: List[str]) -> Set[datetime]:
+    def filepaths_to_datetimes(
+        filepaths: List[str],
+    ) -> Tuple[Set[datetime], Set[datetime]]:
         """
-        utility to convert filepaths to a set of datetimes
+        utility to convert filepaths to a set of datetimes that are archived on
+        the current version of the rail performance manager and a set of
+        datetimes that are archived on other versions
 
         archive filepaths are formatted like
         "s3://<bucket>/<prefix>/YYYY-MM-DD-subway-on-time-performance-v1.parquet"
         """
         # filename is everything after the last "/"
         # first 10 chars are YYYY-MM-DD
         # use strptime to convert to a datetime
-        datetimes = set()
+        this_version = set()
+        old_version = set()
         for filepath in filepaths:
             match = re.search(r"(?P<date>\d{4}-\d{1,2}-\d{1,2})", filepath)
             if match is not None:
-                datetimes.add(
-                    datetime.strptime(match.group("date"), "%Y-%m-%d")
-                )
+                date = datetime.strptime(match.group("date"), "%Y-%m-%d")
+                # check that the version of this file is equal to the current version
+                metadata = object_metadata(filepath).get(S3Archive.VERSION_KEY)
+                if metadata == S3Archive.RPM_VERSION:
+                    this_version.add(date)
+                else:
+                    old_version.add(date)
```

Comment:
i'm not a _big_ fan of doing this call for each file, but it feels necessary?

the other thing i considered was keeping track of it in the index csv, but it was hard to control for errors if something goes wrong mid process.